src/Avalonia.Controls/ComboBox.cs

```diff
@@ -0,0 +1,373 @@
+// Copyright (c) The Avalonia Project. All rights reserved.
+// Licensed under the MIT license. See licence.md file in the project root for full license information.
+
+using System;
+using System.Linq;
+using Avalonia.Controls.Generators;
+using Avalonia.Controls.Presenters;
+using Avalonia.Controls.Primitives;
+using Avalonia.Controls.Shapes;
+using Avalonia.Controls.Templates;
+using Avalonia.Input;
+using Avalonia.LogicalTree;
+using Avalonia.Media;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// A drop-down list control.
+    /// </summary>
+    public class ComboBox : SelectingItemsControl
+    {
+        /// <summary>
+        /// The default value for the <see cref="ItemsControl.ItemsPanel"/> property.
+        /// </summary>
+        private static readonly FuncTemplate<IPanel> DefaultPanel =
+            new FuncTemplate<IPanel>(() => new VirtualizingStackPanel());
+
+        /// <summary>
+        /// Defines the <see cref="IsDropDownOpen"/> property.
+        /// </summary>
+        public static readonly DirectProperty<ComboBox, bool> IsDropDownOpenProperty =
```

Comment:
perhaps we should also change all `DropDown` references in this code to `ComboBox`?

Reply:
This property is also called `IsDropDownOpen` in WPF: https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.combobox.isdropdownopen?view=netframework-4.7.2